frontend/catalyst/jax_tracer.py

```diff
@@ -133,7 +134,9 @@ def traceable_fn(*args, **kwargs):
                                 meas_return_trees_children[i] = counts_tree
                                 meas_return_trees = (
                                     meas_return_trees.make_from_node_data_and_children(
-                                        meas_return_trees.node_data(), meas_return_trees_children
+                                        pytree.PyTreeRegistry(),
```

Comment:
I'm glad this works here :smile: I should study the use case of this new `PyTreeRegistry`  in the JAX PyTree utils.